content/blog/new-graphic-features/index.Rmd

```diff
@@ -0,0 +1,295 @@
+
+### Graphic systems
+
+R currently sports two different systems, one colloquially known as *base* graphics (implemented in the graphics package), and one called *grid* graphics (implemented in the grid package). If you call `plot()` you are most likely to end up using the base graphics system, while grid is used as the basis for e.g. ggplot2. The two systems are largely silos, though effort has been made to allow users to embed base graphics into grid graphics. In RStudio we are mainly invested in the grid graphics system since it powers ggplot2, but by and large, this is all an implementation detail that the user shouldn't care too much about. There might come other graphic systems in the future, and other ways of drawing things on screen or to files also exist outside of the R graphics pipeline (e.g. rgl which allows you to create 3D graphics using an OpenGL/WebGL interface).
+
+### The graphics engine
+
+What unites base and grid graphics and sets them apart from e.g. rgl is that they both call into the same low level C API provided by R. This API is what is called the graphics engine. The graphics engine is responsible for communicating with the graphics devices while also providing a selected range of utility functionality common for both base and grid graphics. It is because of this abstraction that graphics in R is largely decoupled from how it is outputted, be it on screen, in a file, or directly to a printer.
+
+While it sounds nice and neat when it is all laid out like this, the current structure and division has grown out over many years and the boundary between the graphic systems, the graphics engine, and the graphic devices are grey. Still, the design is much more mature than what we see in other languages and as graphics/data-viz developers in R we are pretty spoiled compared to our peers in other languages --- perhaps without really knowing it.
+
+## A fragmented future
+
+With the division of responsibility described above, there are many points in the pipeline that may impose limitations in functionality. The graphic system might not provide the higher-level API to use functionality in the graphics engine, or a graphic device might not provide support for instructions coming in from the graphics engine. While this has mainly been a hypothetical situation prior to R 4.1, it is now the new reality. As users, this is most apparent in the choice of graphic device. After all, you don't expect the graphic system to be capable of something outside of its API, simply because new features were announced for the graphics device. However, if a graphic device lack support you may end up being surprised at what it renders.
```

Comment:
I think this would be a bit easier to follow if you spell out with a concrete example. Like: R 4.1 added support for gradients. This means support was added to graphics engine, the built-in devices (all of theme or just some?), and to grid graphics. What happens now if you want to produce a svg using svglite with ggplot2? What work needs to be done?